File: js/p048.js
"use strict";
const bigInt = require('big-integer');
/*
 * Self powers
 * Problem 48
 * The series, 1^1 + 2^2 + 3^3 + ... + 10^10 = 10405071317.
 * 
 * Find the last ten digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^1000.
 * 
 * Answer: 9110846700
 */

function init() {
	var output = '>>>>> NOT COMPLETED <<<<<';

	var sum = bigInt(0, 10);
	for (var i=1; i<=1000; i++) {
		sum = sum.plus(bigInt(i).pow(i));
	}
	output = sum.toString().substr(-10);
    return output;
}

module.exports.init = init;

File: python/p048.py
"""
	
Project Euler - Problem 48 - Self powers
https://projecteuler.net/problem=48

"""
import timing

def main():
    answer = 0
    limit = 1000
    cut = 10

    answer = sum([n**n for n in range(limit+1)]) - 1
            
    print(answer % 10**cut)

if __name__ == '__main__':
	main()
